lib/flags.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+const masks = [];
+const masksNot = [];
+for (let i = 0; i < 64; i++) {
+  const mask = new Uint64(1).shiftLeft(i);
+  masks.push(mask);
+  masksNot.push(Uint64.not(mask));
+}
+
+class Flags {
+  static from(...args) {
+    if (args.length > 64) {
+      throw new Error('Flags does not support more than 64 values');
+    }
+    const values = new Map(args.map((v, i) => [v, i]));
+
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        if (args[0] instanceof Uint64) {
+          this.value = new Uint64(args[0]);
+        } else {
+          this.value = new Uint64(0);
+          args.forEach(arg => this.set(arg));
+        }
+      }
+
+      has(key) {
+        return values.has(key);
+      }
+
+      get(key) {
+        const v = values.get(key);
+        if (v === undefined) {
+          throw new TypeError(`Flags instance does not have key ${key}`);
```

Comment:
Should we maybe extract the template literal for error message into the function since it is used in multiple places?